pushy/src/main/java/com/turo/pushy/apns/ApnsChannelFactory.java

```diff
@@ -249,4 +253,10 @@ public void close() {
             }
         }
     }
+
+    private static void tryFailureAndLogRejectedCause(final Promise<?> promise, final Throwable cause) {
+        if (!promise.tryFailure(cause)) {
+            log.debug("Could not set failure cause for promise because promise was already done. Rejected cause was {}", cause, cause);
```

Comment:
This logs both a summary of the exception AND a stack trace, which might be overkill. Very open to suggestions here.